var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -68,6 +68,7 @@ class Paraview(CMakePackage, CudaPackage):
                     'catalyst', 'rendering', 'core'),
             description='Build editions include only certain modules. '
             'Editions are listed in decreasing order of size.')
+    variant('vtkm', default=True, description='Build VTKm with ParaView')
```

Comment:
I am against this change. Whether to VTK-m to be enabled by default or not is controlled by the `PARAVIEW_BUILD_EDITION` value

```suggestion
```

Reply:
Oh ok. Maybe this pull request isn't needed anymore. I submitted this before the build_editions were added.. Which build edition can I choose to not include VTKm?

Reply:
@EthanS94 any other than canonical, but then again I recommend against duplicating this logic.

On another topic I think that the other changes are good to have for custom configurations

Reply:
If there's a build edition that disables vtkm, this PR isn't needed. But I don't think lumping vtkm into a build_edition just for spack makes much sense. Build_editions in spack and outside of spack should do the same thing in my opinion.

Reply:
I just tried and it looks like at least `catalyst_rendering` disables vtkm without the logic in this PR.

Reply:
@EthanS94 the build editions set a preset but you can always customize the preset variables value. I am sure that in the future we will like to have in some builds Catalyst + VTKm. Perhaps other maintainers have other opinions on this

cc @danlipsa

Reply:
Sorry I should have been more concrete, the issue with this line is `default=True`, this will enable VTK-m by default which is not something that we want, other changes are just fine

Reply:
Got it! But I get an error without the default:

```console
spack spec paraview                            
Input spec
--------------------------------
paraview

Concretized
--------------------------------
==> Error: either a default was not explicitly set, or 'None' was used [paraview, variant 'vtkm']
```

Reply:
How about this, `default=False` and on the other change only add `if ~vtkm: -DPARAVIEW_USE_VTKM`

Reply:
@vicentebolea I can't see an easy way to have the default be controlled by build_edition and have a separate vtkm variant. Setting the `default=False` is close, but removes vtkm from default canonical builds. I am fine with pointing users to build_editions for now.

If setting the default conditionally was possible, or if it could just default to not having the `vtkm` variant, it would be super easy. Something like:
```py
    variant('vtkm', default=False, description='Build VTKm with ParaView')
    variant('vtkm', default=True, when='build_edition=canonical', description='Build VTKm with ParaView')
```

But unfortunately, as documented in [overriding variants](https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=%22with%20when%22#overriding-variants), it will always default to whatever the last one is.